### ISSUE TITLE
The request signature we calculated does not match the signature you provided

### ISSUE THREAD

**Issue by ColinCee** [zirkelc/aws-sigv4-fetch#4]
Getting a strange error but not sure how to debug the calculated canonical string. I'm trying to authenticate with an internal API Gateway. I'm currently using aws4-axios and axios to make this request so it should work.

```typescript
const signedFetch = createSignedFetcher({
    service: 'execute-api',
    region: 'eu-west-1',
    fetch,
  });

  const response = await signedFetch(
    `https://myhost.com/path?` +
      new URLSearchParams({ $filter: `id eq 123` }),
    {
      method: 'GET',
    },
  );
  // log raw response
  console.log(await response.text());
```

I keep getting this error sensitive info removed:

```
{"message":"The request signature we calculated does not match the signature you provided. Check your AWS Secret Access Key and signing method. Consult the service documentation for details.

The Canonical String for this request should have been
'GET
path
%24filter=id%20eq%20123
host:myhost
x-amz-content-sha256:XXX
x-amz-date:XXX
x-amz-security-token:XXX

host;x-amz-content-sha256;x-amz-date;x-amz-security-token
XXX

The String-to-Sign should have been
'AWS4-HMAC-SHA256
20230731T115020Z
20230731/eu-west-1/execute-api/aws4_request
XXX'
"}
```

I can't share the exact URL but if you have any instructions on how to debug this please let me know

Thanks!

**Comment by zirkelc** [comment]
Hi @ColinCee,
I will take a look at this and get back to you soon!

**Comment by zirkelc** [comment]
Hi @ColinCee 

I was able to reproduce the issue and fix it. I have used the `parseQueryString()` function from `@aws-sdk/querystring-parser` to parse the query string from the provided URL. This function didn't handle URL component encoding/decoding correctly. I changed the implementation to use `new URL().searchParams` and I was able to verify that it works.

The fix is available in the newest v2.1.0. Please let me know if it fixes your issue.

**Comment by ColinCee** [comment]
Amazing thanks for the quick response! I can confirm this is now resolved in v2.1.1

**Comment by zirkelc** [comment]
Thank you for bringing this issue to my attention! :)
